File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/javascript/reverse_string.js
function reverseString(str) {
  // type your code here

  let reversed = '';
  for (let i = str.length - 1; i >= 0; i --){
    reversed += str[i];
  }
  return reversed;
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 'ih'");
  console.log("=>", reverseString("hi"));

  console.log("");

  console.log("Expecting: 'ybabtac'");
  console.log("=>", reverseString("catbaby"));
}

module.exports = reverseString;

// without using built in methods:
// iterate over string, starting from far right index
// create array with those values in order
// return 'reversed' array

File: 01-week-1--starter-algorithms/00-day-1--reverse-a-string/ruby/reverse_string.rb
def reverse_string(str)
  #my code

  #easy solution with built in methods
  # reversed_str = str.reverse
  # return reversed_str

  #solution without using methods, but iteration instead
  reversed_str = ""

  str.chars.each do |character|
    reversed_str = character + reversed_str
  end

  return reversed_str

end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 'ih'"
  puts "=>", reverse_string('hi')

  puts

  puts "Expecting: 'ybabtac'"
  puts "=>", reverse_string('catbaby')


end

#pseudocode
# split string into an array
# reverse the array
# join string into new array
# return new array